src/components/TodoItem.tsx

```diff
@@ -0,0 +1,128 @@
+import classNames from 'classnames';
+import { useState } from 'react';
+import { Todo } from '../types/Todo';
+import { client } from '../utils/fetchClient';
+
+export const TodoItem = ({
+  todo,
+  askTodos,
+  setErrorMessage,
+}: {
+  todo: Todo,
+  askTodos: (url: string) => void
+  setErrorMessage: React.Dispatch<React.SetStateAction<string>>
+}) => {
+  const { id, completed, title } = todo;
+  const [isComplited, setIsComplited] = useState(completed);
+  const [isEdit, setIsEdit] = useState(false);
+  const [temporaryText, setTemporaryText] = useState(title);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleComplited = () => {
```

Comment:
```suggestion
  const handleCompleted = () => {
```

Reply:
In this code, I add debounce for more carefully send ask to server.